cub/test/catch2_test_thread_scan_exclusive_partial.cu

```diff
@@ -0,0 +1,465 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cub/detail/type_traits.cuh>
+#include <cub/thread/thread_scan.cuh>
+#include <cub/util_macro.cuh>
+
+#include <thrust/iterator/constant_iterator.h>
+
+#include <cuda/functional>
+#include <cuda/std/__algorithm/clamp.h>
+#include <cuda/std/functional>
+#include <cuda/std/limits>
+#include <cuda/std/type_traits>
+
+#include <cstring>
+
+#include "c2h/catch2_test_helper.h"
+#include "c2h/extended_types.h"
+#include "c2h/generators.h"
+#include "catch2_test_device_reduce.cuh"
+#include "catch2_test_device_scan.cuh"
+#include "thread_reduce/catch2_test_thread_reduce_helper.cuh"
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+
+constexpr int max_size  = 16;
+constexpr int num_seeds = 3;
+
+/***********************************************************************************************************************
+ * Thread Scan Wrapper Kernels
+ **********************************************************************************************************************/
+
+template <int NumItems, typename In, typename Out, typename Accum, typename ScanOperator>
+__global__ void thread_scan_exclusive_partial_kernel(
+  In d_in, Out d_out, ScanOperator scan_operator, int valid_items, Accum prefix, bool apply_prefix)
+{
+  using value_t  = ::cuda::std::iter_value_t<In>;
+  using output_t = ::cuda::std::iter_value_t<Out>;
+  value_t thread_input[NumItems];
+  _CCCL_PRAGMA_UNROLL_FULL()
+  for (int i = 0; i < NumItems; ++i)
+  {
+    thread_input[i] = d_in[i];
+  }
+  output_t thread_output[NumItems];
```

Comment:
If we should decide to guarantee that `indexes>=valid_items` are not being written to, should we initialize `thread_output` with a magic number - or to make our lives easier - copy the values from `d_in` and then also verify that those items were not overwritten?

Reply:
Copying the items from `d_in` would make life too easy b/c it would not catch if the algorithm was doing exactly that. The magic number is a better test.